src/InstagramClone.Api/Database/InstagramCloneDbContext.cs

```diff
@@ -1,48 +1,72 @@
-﻿using InstagramClone.Api.Entities;
+﻿using InstagramClone.Api.Configuration;
+using InstagramClone.Api.Entities;
 using Microsoft.EntityFrameworkCore;
 
 namespace InstagramClone.Api.Database
 {
     public class InstagramCloneDbContext : DbContext
     {
-        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        public InstagramCloneDbContext(DbContextOptions options) : base(options)
         {
-            optionsBuilder.UseSqlServer("DefaultConnectionString");
-        }
 
+        }
 
+        
 
         protected override void OnModelCreating(ModelBuilder modelBuilder)
         {
+            
+
             modelBuilder.Entity<User>().HasKey(u => u.Id);
             modelBuilder.Entity<Post>().HasKey(p => p.Id);
             modelBuilder.Entity<PostReaction>().HasKey(pr => pr.Id);
 
             // composite Key
-            modelBuilder.Entity<UserFollower>().HasKey(userfollower => new {userfollower.FollowedUserId, userfollower.FollowerId});
+            modelBuilder.Entity<UserFollower>()
+                .HasKey(userfollower => 
+                new {userfollower.FollowedUserId, userfollower.FollowerId});
            
 
             // one to many relationship
             modelBuilder.Entity<Post>().HasOne<User>(post => post.Author)
                 .WithMany( user => user.Posts)
-                .HasForeignKey(fk => fk.AuthorId);
+                .HasForeignKey(fk => fk.AuthorId)
+                .OnDelete(DeleteBehavior.Restrict);
+
+            modelBuilder.Entity<PostReaction>()
+                .HasOne(p => p.Author).
+                WithMany(u => u.Reactions)
+                .HasForeignKey(fk => fk.AuthorId)
+                .OnDelete(DeleteBehavior.Restrict);
+                
 
             modelBuilder.Entity<PostReaction>().HasOne<Post>(reaction => reaction.Post)
                 .WithMany(posts => posts.PostReactions)
-                .HasForeignKey(fk => fk.PostId);
+                .HasForeignKey(fk => fk.PostId)
+                .OnDelete(DeleteBehavior.Restrict);
 
 
             modelBuilder.Entity<UserFollower>()
                 .HasOne<User>(us => us.FollowedUser)
                 .WithMany(u => u.Followed)
-                .HasForeignKey(fk => fk.FollowedUserId);
+                .HasForeignKey(fk => fk.FollowedUserId)
+                .OnDelete(DeleteBehavior.Restrict);
 
             modelBuilder.Entity<UserFollower>()
                 .HasOne(d => d.Follower)
                 .WithMany(m => m.Followers)
-                .HasForeignKey(fk => fk.FollowerId);
+                .HasForeignKey(fk => fk.FollowerId)
+                .OnDelete(DeleteBehavior.Restrict);
+
+            modelBuilder.SeedUser();
+            modelBuilder.SeedPost();
+            modelBuilder.UserFollowerSeed();
```

Comment:
Why is the naming not consistent with the rest.

For the others, you did SeedXXX()

but this one you did XXXSeed()

Please fix

Reply:
Ok, Fixed.